code-challenges.test.js

```diff
@@ -0,0 +1,144 @@
+
+                const fruitArray = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach", "Kiwi"]
+                const filterLetterA = "a"
+                const filterLetterE = "e"
+                
+                expect(wordsWithLetter(fruitArray, filterLetterA)).toEqual(["Mango", "Apricot", "Peach"])
+                expect(wordsWithLetter(fruitArray, filterLetterE)).toEqual(["Cherry", "Blueberry", "Peach"])
+            })
+        })
+
+// b) Create the function that makes the test pass.
+const wordsWithLetter = (words, letter) => {
+    const wordsWithLetter = [];
+    
+    for (let i = 0; i < words.length; i++) {
+      const word = words[i];
+      
+      if (word.includes(letter)) {
+        wordsWithLetter.push(word);
+      }
+    }
+    
+    return wordsWithLetter;
+  }
+
+// Pseudo code:
+// create function named wordsWithLetter it takes two arguments and array and a letter
+// inside the function create an array to hold the return
+// create a for loop to iterate throught the array 
+// create and if statement that will push the word into the new array if it contains the given letter
+// return the new array
+  
+
+// --------------------3) Create a function that takes in an array of 5 numbers and determines whether or not the array is a "full house". A full house is exactly one pair and one three of a kind.
+
+// a) Create a test with expect statements using the variable provided.
+        describe("isFullHouse", () => {
+            it("takes in an array of 5 numbers and determines whether or not the array is a full house.", () => {
+                
+                const hand1 = [5, 5, 5, 3, 3]
+                
+                const hand2 = [5, 5, 3, 3, 4]
+                
+                const hand3 = [5, 5, 5, 5, 4]
+                
+                const hand4 = [7, 2, 7, 2, 7]
+                
+            expect(isFullHouse(hand1)).toEqual(true)
+            expect(isFullHouse(hand2)).toEqual(false)
+            expect(isFullHouse(hand3)).toEqual(false)
+            expect(isFullHouse(hand4)).toEqual(true)
+          
+        })
+        })
+
+        // b) Create the function that makes the test pass.
+const isFullHouse = (numbers) => {
+    const countMap = {};
+  
+    for (let i = 0; i < numbers.length; i++) {
+      const number = numbers[i];
+      countMap[number] = (countMap[number] || 0) + 1;
+    }
+  
+    const counts = Object.values(countMap);
+  
+    return counts.length === 2 && (counts.includes(2) || counts.includes(3));
+  }
+  
+
+// Pseudo code:
+// Create an empty object to store the count of each number in the array.
+// Iterate over each number in the array and increment its count in the object.
+// Get an array of the count values from the object.
+// If the length of the count array is 2 and it contains either a count of 2 or 3, return true (indicating a full house).
+// Otherwise, return false (indicating it's not a full house).
```

Comment:
Nicely done! Great research on using an object to store memory for the count of occurrences. I suggest elaborating some more in your pseudo code to explain how it's adding keys to the object and updating the value at those keys per iteration.